### PR TITLE
Migrate to Dart null safety system

### PR DESCRIPTION
**Pull request by orevial**
Hi,

I propose this pull request to [migrate this plugin to null safety](https://dart.dev/null-safety/migration-guide) as it is the way to go now that Flutter 2 has been released.

Here are the changes I introduced:

* Update SDK dependency version to 2.12.0+
* Update `url_launcher` dependency to the new null-safe version
* Run `dart migrate` on existing code, there is almost no change to the codebase as your code was already null-safe, thank you for that 😄  🎉 
* Fix deprecation warnings
* Format codes

Note that it passes Dart static analysis (`dart analyze`) which means that the package should now get 110/110 pub points on pub.dev 🙂  
 
Thank you for the great work on this plugin and let me know if I can rework something !

### CODE REVIEW

**Comment by orevial** [comment]
Closes #12

**Comment by hmawla** [comment]
This looks perfect

**Comment by orevial** [comment]
@pikaju would you mind having a look at this PR and maybe merge it if it seems OK ?

**Review by rawquesh — COMMENTED** [review]
I can't see changes on the main branch, I am unable to use this package because of dependency error.

**Comment by hmawla** [comment]
> I can't see changes on the main branch, I am unable to use this package because of dependency error.

This is yet to be merged...

**Review by hmawla — APPROVED** [review]

**Comment by amitkot** [comment]
@hmawla @pikaju can this PR be merged?
Thanks!

**Comment by erperejildo** [comment]
anything blocking this PR?

**Comment by hmawla** [comment]
I think the repository author is away for the time being.
You may create a temporary fix by forking this PR and using it directly in your pubspec.yaml like so:

```yaml
  maps_launcher :
    git:
      url: https://github.com/{your-username}/flutter-maps-launcher.git
      ref: master
```

**Comment by amitkot** [comment]
@hmawla I saw you approved the PR, maybe we could get the author to add you as an additional maintainer?

This is a useful package, I would love for it to stay maintained.

**Comment by erperejildo** [comment]
Working with in the meantime:
```
maps_launcher:
    git:
      url: https://github.com/orevial/flutter-maps-launcher.git
      ref: null-safety-migration
```

**Comment by mendoxe** [comment]
Any news regarding this topic?

**Comment by pikaju** [comment]
Damn it, I really need email notifications. Sorry for the wait guys. I will get a new version published ASAP. Thank you for your contribution!

**Pull request merged**
